### ISSUE TITLE
Exception on `rake release` when login is required

### ISSUE THREAD

**Issue by paul** [bkuhlmann/gemsmith#2]
## Expected Behavior

`rake release` to work when login was required

## Actual Behavior

It didn't work

## Steps to Recreate

1. Not be logged in to rubygems.org. (Delete or rename `~/.gem/credentials`)
1. `rake release` a gem
1. 💥 

```
$ rake release
/Users/rando/.gem/ruby/2.4.1/gems/rake-11.2.2/lib/rake/ext/fixnum.rb:4: warning: constant ::Fixnum is deprecated
Cleaned gem artifacts.
Updated gem documentation.
sliding_partition 0.5.0 built to pkg/sliding_partition-0.5.0.gem.
https://github.com/paul/sliding_partition.git
Counting objects: 1, done.
Writing objects: 100% (1/1), 212 bytes | 0 bytes/s, done.
Total 1 (delta 0), reused 0 (delta 0)
To https://github.com/paul/sliding_partition.git
 * [new tag]         v0.5.0 -> v0.5.0
What is your "https://rubygems.org" login? psadauskas@gmail.com
What is your "https://rubygems.org" password? {secret}
rake aborted!
NameError: uninitialized constant Gemsmith::Authenticators::RubyGems::OpenSSL
Did you mean?  Open3
/Users/rando/.gem/ruby/2.4.1/gems/gemsmith-7.7.0/lib/gemsmith/authenticators/ruby_gems.rb:35:in `configure_client'
/Users/rando/.gem/ruby/2.4.1/gems/gemsmith-7.7.0/lib/gemsmith/authenticators/ruby_gems.rb:18:in `initialize'
/Users/rando/.gem/ruby/2.4.1/gems/gemsmith-7.7.0/lib/gemsmith/credentials.rb:54:in `new'
/Users/rando/.gem/ruby/2.4.1/gems/gemsmith-7.7.0/lib/gemsmith/credentials.rb:54:in `create'
/Users/rando/.gem/ruby/2.4.1/gems/gemsmith-7.7.0/lib/gemsmith/rake/release.rb:29:in `push'
/Users/rando/.gem/ruby/2.4.1/gems/gemsmith-7.7.0/lib/gemsmith/rake/release.rb:38:in `publish'
/Users/rando/.gem/ruby/2.4.1/gems/gemsmith-7.7.0/lib/gemsmith/rake/tasks.rb:46:in `block in install'
/Users/rando/.gem/ruby/2.4.1/gems/rake-11.2.2/exe/rake:27:in `<top (required)>'
```

(Also, gemsmith should really mask or not echo the password, similar to how the `gem login` command works.)

**Comment by paul** [comment]
Also, we've discussed this before, but its annoying that you can't `gem login` and then retry gemsmith release:

```
$ rake release
/Users/rando/.gem/ruby/2.4.1/gems/rake-11.2.2/lib/rake/ext/fixnum.rb:4: warning: constant ::Fixnum is deprecated
Cleaned gem artifacts.
Updated gem documentation.
sliding_partition 0.5.0 built to pkg/sliding_partition-0.5.0.gem.
Duplicate tag exists: v0.5.0.
```

**Comment by bkuhlmann** [comment]
Thanks for catching that. I've got a fix for this [here](https://github.com/bkuhlmann/gemsmith/commit/8dbb82fb98b5). The issue was with the independent loading of those modules within the context of Rake.

I'm working on a fix for masking the password too.

I'll definitely have a new release out by this weekend for you.

**Comment by bkuhlmann** [comment]
I've fixed the issue with masking of the password prompt as well. I switched to using Thor instead of Bundler for this. I have a [release](https://github.com/bkuhlmann/gemsmith/compare/release?expand=1) branch with these changes that I'll push out this weekend.

The last issue, with the existence of the duplicate tag issue, is actually with the [Milestoner](https://github.com/bkuhlmann/milestoner) gem which Gemsmith uses under the covers. The safest way to handle this is by skipping that check if the current version being released is identical to what is tagged locally and on the remote.

More news soon!

**Comment by bkuhlmann** [comment]
I've released [v9.4.0](https://github.com/bkuhlmann/gemsmith/releases/tag/v9.4.0) which addresses the issues above except the duplicate tag issue. I have somewhat of fix in place for the Milestoner gem but it doesn't, fully, solve the problem due the complexity of checking local and remote tags and making sure they are identical or different (and if they need to be rebuilt, etc). I need to spend more time with this before releasing a proper fix for the Milestoner gem.

At least with Gemsmith, this issue is resolved.
